app/models/spree/order_decorator.rb

```diff
@@ -276,12 +276,19 @@ def tax_adjustment_totals
       tax_rates = adjustment.tax_rates
       tax_rates_hash = Hash[tax_rates.collect do |tax_rate|
         tax_amount = tax_rates.one? ? adjustment.included_tax : tax_rate.compute_tax(adjustment.amount)
-        [tax_rate.amount, tax_amount]
+        [tax_rate, tax_amount]
       end]
       hash.update(tax_rates_hash) { |_tax_rate, amount1, amount2| amount1 + amount2 }
     end
   end
 
+  def price_adjustment_totals
```

Comment:
So, is this method called from the view? I'm not fully grasp the fix.

Reply:
the method is called from the invoice2.html view and the tax_rate report. for those views returning a Hash of `tax_rate_amount (decimal) => tax_aggregate_amount (decimal)` was all that was needed, but the `price_adjustment_totals` method called from the admin order show view expects a Hash of `tax_name (string) => tax_aggregate_amount (Money)`, so I made this method return a Hash of `tax_rate (TaxRate) => tax_aggregate_amount (decimal)` so we can call `TaxRate#name` in `price_adjustment_totals` and `TaxRate#amount` from the other places
(we return the whole object and delegate the call to its attribute to places where the method is called)